### PR TITLE
Kuka with Robotiq 3 finger gripper

### PR DESCRIPTION
**Pull request by christian-rauch**
WIP: This needs to be verified by comparing FK with image.

Robot model that connects the Kuka LWR with the Robotiq 3-finger gripper.

## Measurements

The dimensions are taken from the [official specification](https://assets.robotiq.com/production/support_documents/document/online/HTML5_20180131.zip/HTML5/Default.htm#3-Finger-Instructions-Manual-06-2017/6.%20Specifications.htm#6.6_Couplings%3FTocPath%3D6.%2520Specifications|6.6%2520Couplings|_____0) and by manual measurement.

The parts involved are:
img_20181021_175931

1. kuka thread connector, length: 22mm
2. aluminium thread, length: 8mm
3. aluminium ring, length: 12mm
4. robotiq coupling ring, length: 15.9mm
5. robotiq coupling plate, length: 3.175mm
6. distance to palm origin: 41.275mm

The accumulated offset in z-directoin is 102.35mm.

## URDF
Rendered model:
lwr_robotiq_model

Note that in the palm mesh, the connectors are on the wrong site. In the original CAD (`AGS-CPL-175_20171219.STEP`) and on the real palm, the power and data connectors are on the side with the single finger (thumb). In the palm mesh (`palm.STL`) of package `robotiq_s_model_visualization`, the connectors are on the side with the two fingers. However, the origin is the same.

The final transformation between the last arm link and the palm is:
```
$ rosrun tf tf_echo lwr_arm_7_link robotiq_palm
At time 0.000
- Translation: [0.000, 0.000, 0.102]
- Rotation: in Quaternion [0.542, -0.455, -0.455, 0.542]
            in RPY (radian) [1.571, -0.000, -1.396]
            in RPY (degree) [90.000, -0.000, -80.000]
```

Does this seem correct?

I approximated the parts between last arm link and robotiq coupling ring with cylinders. Do we have the meshes of these 3 parts?

### CODE REVIEW

File: robots/lwr_robotiq_3finger.urdf.xacro [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  <xacro:arg name="use_gazebo" default="1"/>
+  <xacro:arg name="use_transmission" default="1"/>
+  <xacro:arg name="collision_simple" default="1"/>
+  <xacro:arg name="visual_simple" default="0"/>
+
+  <!-- KUKA LWR4 arm -->
+  <xacro:include filename="$(find kuka_lwr4_description)/robots/kuka_lwr_arm.urdf.xml"/>
+
+  <!-- Robotiq 3 finger hand -->
+  <xacro:include filename="$(find robotiq_s_model_visualization)/cfg/s-model_articulated_macro.xacro" />
```
Comment (wxmerkt) [inline robots/lwr_robotiq_3finger.urdf.xacro]: Try to use the newer, ``kinetic-devel`` head here: the package is now renamed to be ``robotiq_3f_gripper_visualization``.

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic, I therefore assumed that it is not ready yet.
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Huskies?

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic, I therefore assumed that it is not ready yet.
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Huskies?

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic. I therefore assumed that it is not ready yet (http://repositories.ros.org/status_page/ros_kinetic_default.html?q=robotiq).
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Huskies?

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic. I therefore assumed that it is not ready yet (http://repositories.ros.org/status_page/ros_kinetic_default.html?q=robotiq).
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Huskies?

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic. I therefore assumed that it is not ready yet (http://repositories.ros.org/status_page/ros_kinetic_default.html?q=robotiq).
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Husky?

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic. I therefore assumed that it is not ready yet (http://repositories.ros.org/status_page/ros_kinetic_default.html?q=robotiq).
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Husky?

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic. I therefore assumed that it is not ready yet (http://repositories.ros.org/status_page/ros_kinetic_default.html?q=robotiq).
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Husky?

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic. I therefore assumed that it is not ready yet (http://repositories.ros.org/status_page/ros_kinetic_default.html?q=robotiq).
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Husky?

Comment (christian-rauch) [reply robots/lwr_robotiq_3finger.urdf.xacro]: `jade-devel` is still the default branch and the `robotiq` package hasn't been released yet to kinetic. I therefore assumed that it is not ready yet (http://repositories.ros.org/status_page/ros_kinetic_default.html?q=robotiq).
Are you successfully using the newer `kinetic-devel` branch on the dual-arm Husky?

**Comment by christian-rauch** [comment]
The dimensions of the coupling seem to fit.

Overlayed the FK state on top of the camera image:
screenshot from 2018-10-23 14-55-24

3D view:
peek 2018-10-23 14-55

**Comment by wxmerkt** [comment]
Awesome, thank you for confirming! Do you mind changing to ``robotiq_3f_gripper_visualization``? It is the kinetic-devel branch of robotiq.

**Pull request merged**